### ISSUE TITLE
JSon compiler and Generics

### ISSUE THREAD

**Issue by careri** [Starcounter/Starcounter#3640]
Hi!

Ran into some trouble with using a generic type for the partial code-behind class:
````
partial class WorkstationPage : Partial<string, string>, IBound<WorkstationViewModel> {}
````

I solved it using this ugly solution. Maybe a fix in the build task is possible?
````
// Dummy class to dispose of direct inherit of generic class
public abstract class PartialStringString : Partial<string, string> { }

partial class WorkstationPage : PartialStringString, IBound<WorkstationViewModel> {}
    
````

**Comment by per-samuelsson** [comment]
Hi @careri 

We are just in the middle of [doing some work on the code-behind parser](#3439). I'll see if I can somehow improve the experience of this in that. Let us get back to you next week with more info.

**Comment by per-samuelsson** [comment]
I verified this and could reproduce the flaw in the current version (using the Mono-based code-behind parser). I also verified that I've properly fixed it with #3439. Once we merge that, this issue will be fixed.

**Comment by per-samuelsson** [comment]
>  I also verified that I've properly fixed it with #3439. Once we merge that, this issue will be fixed.

Done. Issue fixed.
